fix: update NodeFeatureDiscovery to work with the latest OCP

e0ne opened the pull request.

noama-nv commented:
@e0ne i think we should link to red hat NFD installation guide, keep only the changes we need in the pci section

image can change and cause issues if we specify and users will just copy paste it

e0ne commented:
> @e0ne i think we should link to red hat NFD installation guide, keep only the changes we need in the pci section
> 
> image can change and cause issues if we specify and users will just copy paste it

I updated an image as mentioned in a NFD guide and added a link to official documentaion
